### DISCUSSION TITLE
How to override the data class name of view in .drift files.

### DISCUSSION THREAD

**Discussion by masahiro-ikemoto** [simolus3/drift#2912] [Q&A] [answered]
The data class name of a table can be overridden by AS, but can the data class name of a view be overridden?

```CREATE TABLE Categories (...) AS Category;``` <-OK
```CREATE VIEW Categories AS SELECT ... [AS Category];``` <- HOW?

**Comment by simolus3** [comment, marked as answer]
At the moment, custom names for views in drift files are not supported due to the ambiguities in syntax.

You can use existing row classes by writing `CREATE VIEW Categories WITH MyCategoryClass AS SELECT ...`, but `AS` is not supported.

**Reply by masahiro-ikemoto** [reply]
Thanks for your response. I found that it is not supported.
Are there any plans to support it?

**Reply by simolus3** [reply]
I've added support for that in 29303100b80d41b92f48b4bebff8b4a39075e320, so the next drift version will support it.
